src/compiler/checker.ts

```diff
@@ -1107,6 +1107,24 @@ namespace ts {
             return diagnostic;
         }
 
+        function addDeprecatedSuggestion(location: Node, declarations: Node[], deprecatedEntity: string) {
+            const diagnostic = createDiagnosticForNode(location, Diagnostics._0_is_deprecated, deprecatedEntity);
+            for (const declaration of declarations) {

```

Comment:
It seems like you semantically want to find the first encounter of a deprecated tag and add diagnostics info to it. I would rather extract finding an element based on predicate into a standalone function (aka `Array.find()` from ES6) and then do an `if (deprecatedTag)` check with a return value of that function.